tools/manifest-templator/manifest-templator.go

```diff
@@ -0,0 +1,216 @@
+			// spaces is is indention - 2, because we want to have 2 spaces less for being able to start an array
+			spaces := strings.Repeat(" ", indention-2)
+			if strings.HasPrefix(line, "apiGroups") {
+				// spaces + array start
+				out.WriteString(spaces + "- " + line + "\n")
+			} else {
+				// 2 more spaces
+				out.WriteString(spaces + "  " + line + "\n")
+			}
+		}
+	}
+	return out.String()
+}
+
+func marshallObject(obj interface{}, writer io.Writer) error {
+	jsonBytes, err := json.Marshal(obj)
+	check(err)
+
+	var r unstructured.Unstructured
+	if err := json.Unmarshal(jsonBytes, &r.Object); err != nil {
+		return err
+	}
+
+	// remove status and metadata.creationTimestamp
+	unstructured.RemoveNestedField(r.Object, "template", "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "status")
+
+	jsonBytes, err = json.Marshal(r.Object)
+	if err != nil {
+		return err
+	}
+
+	yamlBytes, err := yaml.JSONToYAML(jsonBytes)
+	if err != nil {
+		return err
+	}
+
+	// fix templates by removing quotes...
+	s := string(yamlBytes)
+	s = strings.Replace(s, "'{{", "{{", -1)
+	s = strings.Replace(s, "}}'", "}}", -1)
+	yamlBytes = []byte(s)
+
+	_, err = writer.Write([]byte("---\n"))
+	if err != nil {
+		return err
+	}
+
+	_, err = writer.Write(yamlBytes)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getCNA(data *templateData) {
+	writer := strings.Builder{}
+
+	// Get CNA Deployment
+	cnadeployment := components.GetDeployment(
+		data.ContainerPrefix,
+		data.ContainerTag,
+		data.ImagePullPolicy,
+	)
+	err := marshallObject(cnadeployment, &writer)
+	check(err)
+	deployment := writer.String()
+
+	// Get CNA DeploymentSpec for CSV
+	writer = strings.Builder{}
+	err = marshallObject(cnadeployment.Spec, &writer)
+	check(err)
+	deploymentSpec := fixResourceString(writer.String(), 12)
+
+	// Get CNA ClusterRole
+	writer = strings.Builder{}
+	clusterRole := components.GetClusterRole()
+	marshallObject(clusterRole, &writer)
+	clusterRoleString := writer.String()
+
+	// Get the Rules out of CNA's ClusterRole
+	writer = strings.Builder{}
+	cnarules := clusterRole.Rules
+	for _, rule := range cnarules {
+		err := marshallObject(rule, &writer)
+		check(err)
+	}
+	rules := fixResourceString(writer.String(), 14)
+
+	// Get CNA CRD
+	writer = strings.Builder{}
+	crd := components.GetCrd()
+	marshallObject(crd, &writer)
+	crdString := writer.String()
+
+	// Get CNA CR
+	writer = strings.Builder{}
+	cr := components.GetCR()
+	marshallObject(cr, &writer)
+	crString := writer.String()
+
+	cnaData := operatorData{
+		Deployment:        deployment,
+		DeploymentSpec:    deploymentSpec,
+		ClusterRoleString: clusterRoleString,
+		Rules:             rules,
+		CRD:               crd,
+		CRDString:         crdString,
+		CRString:          crString,
+	}
+	data.CNA = &cnaData
+}
+
+func main() {
+	csvVersion := flag.String("csv-version", "0.0.1", "")
```

Comment:
defaults are already in the shell script. can we drop it from one of those places?